[Kotlin Warnings As Errors] Woo Commerce Module - Resolve Type/Cast/When & Other Warnings

ParaskP7 opened the pull request:
<!-- Remember about a good descriptive title. -->

Parent: #6816
Partially Closes: #6819

<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR resolves all `Type/Cast/When` & `Other` related warnings for the `woocommerce` module:

- 9 warnings x `type`
    - 1 warning x `Unsafe use of a nullable receiver of type Xyz?`
    - 1 warning x `Returning type parameter has been inferred to Nothing implicitly because Nothing is more specific than specified expected type. Please specify type arguments explicitly in accordance with expected type to hide this warning. Nothing can produce an exception at runtime. See KT-36776 for more details.`
    - 1 warning x `Type mismatch: inferred type is Xyz? but Xyz was expected`
    - 1 warning x `Elvis operator (?:) always returns the left operand of non-nullable type Boolean`
    - 5 warnings x `Unnecessary safe call on a non-null receiver`
- 6 warnings x `cast`
    - 3 warnings x `Smart cast to 'Xyz' is deprecated, because 'itemView' is a property declared in base class from different module`
    - 2 warnings x `Unchecked cast: Any to T`
    - 1 warning x `No cast needed`
- 10 warnings x `when`
    - 7 warnings x `Non exhaustive 'when' statements`
    - 3 warnings x `'when' is exhaustive so 'else' is redundant here`
- 18 warnings x `other`
    - 1 warnings x `This is a delicate API and its use requires care. Make sure you fully read and understand documentation of the declaration that is marked as a delicate API.`
    - 9 warnings x `Parameter 'xyz' is never used`
    - 3 warnings x `Parameter 'xyz' is never used, could be renamed to _`
    - 2 warnings x `Name shadowed: payload`
    - 1 warning x `Property would not be serialized into a 'Parcel'. Add '@IgnoredOnParcel' annotation to remove the warning`
    - 2 warnings x `Parcelize annotations from package 'kotlinx.android.parcel' are deprecated. Change package to 'kotlinx.parcelize'`

Warnings Resolution List:

1. `type`
    - [Resolve to unsafe use of nullable receiver type warning.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/0777041f36ace336bdba88a2298e9c02d8ea144b)
    - [Resolve type parameter inferred to nothing type warning.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/b045e1beb1aaa960cab5216ff385933f69e23579)
    - [Resolve type mismatch type warning.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/504406316e4b331b65c5be5d2d8e881f75a9356e)
    - [Resolve elvis operator type warning.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/f6596e991c98c05767289169afd4d0a4dbde6a97)
    - [Resolve unnecessary safe call type warnings.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/0d3fdef628c3c425078436095337046a6641484c)
2. `cast`
    - [Resolve item view smart cast is deprecated cast warnings.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/791d8a4184290d9dc7fd10eb272134bf5cb7f082)
    - [Resolve unchecked cast any to list cast warning.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/6a7c69f75eda64aa5524c4bfbc490339f5e16035)
3. `when`
    - [Resolve non exhaustive when statements when warnings.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/5539eb683c87bbfd2d2cdfc6b2f128a7151bf34e)
    - [Resolve else is redundant when warnings.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/ccda8a62213ca2b9e604f4cd5851452dc60c7784)
4. `other`
    - [Resolve global scope delicate coroutines api warning.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/9214ccedb35c016395865b97bcfe375d4f837db7)
    - [Resolve parameter is never used warnings.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/c57aa41d80204a05e99c5103e5043bb5e9c15239)
    - [Resolve parameter is never and could be renamed to _ warnings.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/1795519b341ffe53edfa58f1b6bdd15c1ccef708)
    - [Resolve name shadowed warnings.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/bdbdb6624288cf6b34a181e8094d55e642414adc)
    - [Resolve kotlinx parcelize deprecated annotations warnings.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/781e1b19ec04550959fe7f9816aefe56d4663cde)

Warnings Suppression List:

1. `cast`
    - [Suppress unchecked cast any to t cast warning for input field.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/e6ad5166b4303d63060add114c68f6657839b453)
    - [Suppress no cast needed cast warning.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/eaaf0ead481c9ad94323ec6aa7a58e1c3cb242d5)
2. `other`
    - [Suppress parameter is never used warnings.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/7210e5362beb7cf6108110ef3360fe72e7336a20)
    - [Suppress property serialized into parcel warning.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/7f6ec8b554a09d755d92a6bb5c7e1dbf28bc4984)

Refactor List:

1. [Remove unused sequence contains instance of extension function.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/aa6d0e1a516f69f369ec8e58d979ebc42a0f7b4d)
4. [Optimize imports for product detail view model.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/6db22365a2765a84cff03423105d0a68e2f3ed19)

In addition to the above warning resolution list, the below `Detekt` warning got resolved as well:

1. [Suppress complex method detekt warning for product detail vm.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/a0c6a622278b7be17246a8254be92043d73b74bd)

-----

Extras new warnings that got resolved after pulling new code from `trunk`:

1. [Resolve get color deprecated warnings.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/456531b11fd30731e859d14dd6be8aa273282066)
2. [Remove redundant experimental coroutines api annotations.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/8f71ff1ac641d8157b454e7f87360595340e397f)

Merge conflict related:

1. [Remove unnecessary detekt baseline entries.](https://github.com/woocommerce/woocommerce-android/pull/6865/commits/70cb1ff957420d246a300a5753dce260905bfd53)

-----

PS.1: @atorresveiga I added you as the main reviewer, that is, in addition to the @woocommerce/owl-team team itself, but randomly, since I just wanted someone from the `Woo Android` team to sing-off on that change for WCAndroid as well. PS: I am going to randomly add more of you in those PRs just so you become more aware of this change and how close we are on enabling `allWarningsAsErrors` by default everywhere. 🥇 

PS.2: With that and the [deprecated](https://github.com/woocommerce/woocommerce-android/pull/6850) sibling PR merged to `trunk` there are going to be no such warnings left to be resolved for the `main` source of this module. For the `test` source of this module we still have `174` such warnings to solve. 🎉

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->

- Verify that all the CI checks are successful.
- Smoke test the app and verify everything works as expected.

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

ParaskP7 committed:
Analysis: Resolve to unsafe use of nullable receiver type warning

Warning Message: "Unsafe use of a nullable receiver of type Date?"

Replacing with safe (?.) call is the recommended action to resolve this
kind of type warnings.

ParaskP7 committed:
Analysis: Resolve type parameter inferred to nothing type warning

Warning Message: "Returning type parameter has been inferred to Nothing
implicitly because Nothing is more specific than specified expected
type. Please specify type arguments explicitly in accordance with
expected type to hide this warning. Nothing can produce an exception at
runtime. See KT-36776 for more details."

Replacing the safe 'xyz?.let { ... }' call with the plain
'if (xyz != null) { ... }' is the recommended action to resolve
this kind of type warnings.

ParaskP7 committed:
Analysis: Resolve type mismatch type warning

Warning Message: "Type mismatch: inferred type is Date? but Date was
expected"

Replacing the plain 'if (xyz != null) { ... }' with the safe
'xyz?.let { ... }' call with  is the recommended action to resolve
this kind of type warnings.

ParaskP7 committed:
Analysis: Resolve elvis operator type warning

Warning Message: "Elvis operator (?:) always returns the left operand of
non-nullable type Boolean"

Removing the useless elvis operator is the recommended action to resolve
this kind of type warnings.

ParaskP7 committed:
Analysis: Resolve unnecessary safe call type warnings

Warning Message: "Unnecessary safe call on a non-null receiver of type
Xyz?. This expression will have nullable type in future releases"

Replacing with dot call is the recommended action to resolve this kind
of type warnings.

ParaskP7 committed:
Analysis: Resolve item view smart cast is deprecated cast warnings

Warning Message: "Smart cast to 'Xyz' is deprecated, because 'itemView'
is a property declared in base class from different module"

Using 'context' and a casted 'propertyView' is the recommended action
to resolve this kind of cast warnings.

ParaskP7 committed:
Analysis: Suppress unchecked cast any to t cast warning for input field

Warning Message: "Unchecked cast: Any to T"

This cast warning is suppressed instead of being resolved since a
resolution would require updating the abstract 'InputField' class as
well, which might be best to ignore as out of scope for now so as to not
introduce any breaking changes to this functionality overall.

ParaskP7 committed:
Analysis: Resolve unchecked cast any to list cast warning

Warning Message: "Unchecked cast: Any? to List<Long>"

Update '(event.data as? List<Long>) ?: emptyList()'
'(event.data as? List<*>) ?: emptyList<Long>()' is the recommended
action to resolve this kind of cast warnings.

ParaskP7 committed:
Analysis: Suppress no cast needed cast warning

Warning Message: "No cast needed"

Explanation: As per the already existing comment, casting added to work
around IDE bug of not recognizing the nested class type.

ParaskP7 committed:
Analysis: Resolve non exhaustive when statements when warnings

Warning Message: "Non exhaustive 'when' statements"

Adding an else branch or remaining branches is the recommended
action to resolve this kind of when warnings.

1) You will notice that I added remain branches wherever possible, while
only added an else branch to the 'MainPresenter.onOrderChanged(...)' and
'PrivacySettingsPresenter.onAccountChanged(...)' when statements because
otherwise those will be blown up with cases that are meant to be ignored
anyway.
2) Also, you will notice that I optimized the imports on all those when
statements and the imports they were using because one or no import is
always better than having multiple imports for no particular reason.
Also, having the extra information like 'RequestResult.SUCCESS' instead
of plain 'SUCCESS' is usually better for readability purposes as the
reader will immediately understand that this 'SUCCESS' is
'RequestResult' related, without the need to navigate to the 'SUCCESS'
in order to understand when that is coming from.

ParaskP7 committed:
Analysis: Resolve else is redundant when warnings

Warning Message: "'when' is exhaustive so 'else' is redundant here"

Removing the else branch is the recommended action to resolve this kind
of when warnings.

ParaskP7 committed:
Analysis: Resolve global scope delicate coroutines api warning

Warning Message: "This is a delicate API and its use requires care.
Make sure you fully read and understand documentation of the declaration
that is marked as a delicate API."

Adding the '@OptIn(DelicateCoroutinesApi::class)' annotation for the
'initGlideCache(...)' function is the recommended action to resolve this
'GlobalScope' warnings.

ParaskP7 committed:
Refactor: Remove unused sequence contains instance of extension function

ParaskP7 committed:
Analysis: Resolve parameter is never used warnings

Warning Message: "Parameter 'xyz' is never used"

Removing the parameter 'xyz' that is never used is the recommended
action to resolve this kind of warnings.

ParaskP7 committed:
Analysis: Suppress parameter is never used warnings

Warning Message: "Parameter 'xyz' is never used"

Explanation: For 'Event Bus' related functions this warning should be
suppressed instead. This is because this is how the 'Event bus'
mechanism works as it need the parameter to be present on a function,
even if that parameter is unused.

ParaskP7 committed:
Analysis: Resolve parameter is never and could be renamed to _ warnings

Warning Message: "Parameter 'xyz' is never used, could be renamed to _"

Renaming the parameter 'xyz' that is never used to '_' or removing it
completely is the recommended action to resolve this kind of warnings.

ParaskP7 committed:
Analysis: Resolve name shadowed warnings

Warning Message: "Name shadowed: xyz"

Renaming the shadowed and/or the shadowing 'xyz' name is the recommended
action to resolve this kind of deprecated warnings.

ParaskP7 committed:
Analysis: Suppress property serialized into parcel warning

Warning Message: "Property would not be serialized into a 'Parcel'.
Add '@IgnoredOnParcel' annotation to remove the warning"

This deprecated warning is suppressed instead of being resolved since a
resolution would require a proper solution and/or verification.

ParaskP7 committed:
Analysis: Resolve kotlinx parcelize deprecated annotations warnings

Warning Message: "Parcelize annotations from package
'kotlinx.android.parcel' are deprecated. Change package to
'kotlinx.parcelize'"

Renaming the deprecated 'kotlinx.android.parcel.Parcelize' with the new
'kotlinx.parcelize.Parcelize' import is the recommended action to
resolve this kind of deprecated warnings.

ParaskP7 committed:
Analysis: Suppress complex method detekt warning for product detail vm

ParaskP7 committed:
Refactor: Optimize imports for product detail view model

Also, with this change 32 entries are removed from the Detekt related
'baseline.xml' file as they are no longer needed. The imports were
optimized on all those main and test source related files that were
violating the (no) wildcard imports rule.

ParaskP7 committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce-android into analysis/woo-commerce-type-cast-when-warnings

 Conflicts:
	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/main/MainPresenter.kt
	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt
	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductSelectionListFragment.kt
	config/detekt/baseline.xml

ParaskP7 committed:
Analysis: Remove redundant experimental coroutines api annotations

Also, as part of this commit the more focus
'@OptIn(ExperimentalCoroutinesApi::class)' annotation was added for
'CreateUpdateOrder.invoke(...)' as it is better to mark that locally.

ParaskP7 committed:
Analysis: Resolve get color deprecated warnings

Warning Message: "'getColor(Int): Int' is deprecated. Deprecated in
Java"

Replacing 'resources.getColor(R.color.xyz)' with
'ContextCompat.getColor(requireContext(), R.color.xyz))' is the
recommended action to resolve this kind of deprecated warnings.

ParaskP7 committed:
Analysis: Remove unnecessary detekt baseline entries

These entries were readded to 'baseline.xml' after an unsuccessful merge
conflict.

atorresveiga commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt:
```diff
@@ -612,33 +607,39 @@ class ProductDetailViewModel @Inject constructor(
                 eventName = AnalyticsEvent.PRODUCT_VARIATION_ATTRIBUTE_ADDED_BACK_BUTTON_TAPPED
                 hasChanges = hasAttributeChanges()
             }
+            is ProductExitEvent.ExitLinkedProducts -> Unit // Do nothing
```
Np: Won't it be readable to add an `else` branch here?

ParaskP7 replied:
👋 @atorresveiga and thank you for the comment! 🙇 

Here is my thought process and why I decide in favor or again the `else` branch:
- If the `when(...)` statement has already more than half of the branches specified, then I'll go for adding the missing branches instead of a generic `else` branch. In general, no matter if it is more readable on not, I would prefer if the missing branches was added explicitly so that the readers would be aware of all the branches and act on that more thoughtfully. But, sometimes I agree it does make sense and I'll go with the `else` branch.
- If the `when(...)` statement has already a few branches, but missing most of them, then I'll go for adding the `else` branch instead of adding all the missing branches. The `MainPresenter.onOrderChanged(...)` is a good example of that, where I adding the `else` branch instead of adding all the missing branches. This is because the already existing branches specified was `2`, and the missing branches was `10`, thus it made sense. An even better example is the `PrivacySettingsPresenter.onAccountChanged(...)` where there existing only `1` branch and the missing ones was `37`.

Thus, I chose to proceed with one solution over the other, based on this arbitrary rule of mine at this point, but no matter what to resolve the warning in the end. Does the above make sense? 😅

atorresveiga commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/reviews/ReviewDetailViewModel.kt:
```diff
@@ -98,15 +98,17 @@ class ReviewDetailViewModel @Inject constructor(
                     }
-                    ERROR -> triggerEvent(ShowSnackbar(R.string.wc_load_review_error))
+                    RequestResult.ERROR -> triggerEvent(ShowSnackbar(R.string.wc_load_review_error))
+                    RequestResult.API_ERROR -> Unit // Do nothing
```
Np: same comment as above. It is not something that needs to be changed, but I'm curious if having an else branch here will help with readability. Another option here to don't duplicate the `-> Unit` could be :
```
RequestResult.API_ERROR, RequestResult.RETRY -> Unit
```

ParaskP7 replied:
👋 @atorresveiga and thank you for the comment! 🙇

Same [response](https://github.com/woocommerce/woocommerce-android/pull/6865#discussion_r916658368)! 😅 

> Another option here to don't duplicate the -> Unit could be :
> 
> RequestResult.API_ERROR, RequestResult.RETRY -> Unit

That's an interesting idea and thanks for bringing this up. It is just that I usually go with always having every branch in its own as I feel that having two branches like that, split by a comma (`, `) can create confusion, hide branches from you if you are not careful enough, and that can even create problems sometimes. I for sure experienced that a few times already on code reviews and bugs that occurred due to the fact that the author didn't notice that happening. Thus, nowadays, I am always voting in favor of the following instead:

```
RequestResult.API_ERROR -> Unit
RequestResult.RETRY -> Unit
```

Think about the `Unit` as a function. Sometime people have problem when duplicating code like this:

```
WCOrderAction.FETCH_ORDERS -> {
    // we just fetched the order list or an order's status changed, so re-check the unfilled orders count
    WooLog.d(WooLog.T.ORDERS, "Order status changed, re-checking unfilled orders count")
    fetchUnfilledOrderCount()
}
WCOrderAction.UPDATE_ORDER_STATUS -> {
    // we just fetched the order list or an order's status changed, so re-check the unfilled orders count
    WooLog.d(WooLog.T.ORDERS, "Order status changed, re-checking unfilled orders count")
    fetchUnfilledOrderCount()
}
```

So they chose the alternative instead:

```
WCOrderAction.FETCH_ORDERS, WCOrderAction.UPDATE_ORDER_STATUS -> {
    // we just fetched the order list or an order's status changed, so re-check the unfilled orders count
    WooLog.d(WooLog.T.ORDERS, "Order status changed, re-checking unfilled orders count")
    fetchUnfilledOrderCount()
}
```

I actually prefer a third solution, the below:

```
WCOrderAction.FETCH_ORDERS -> function()
WCOrderAction.UPDATE_ORDER_STATUS -> function()

private fun function() {
    // we just fetched the order list or an order's status changed, so re-check the unfilled orders count
    WooLog.d(WooLog.T.ORDERS, "Order status changed, re-checking unfilled orders count")
    fetchUnfilledOrderCount()
}
```

Does it help you understand why I chose this path? For sure anyone can chose their own way of doing things and for their own reasons, this is highly subjective. I don't mind too much anyway... 😅

atorresveiga commented on WooCommerce/src/main/kotlin/com/woocommerce/android/widgets/sectionedrecyclerview/Section.kt:
```diff
@@ -77,10 +74,10 @@ abstract class Section(sectionParameters: SectionParameters) {
-                LOADED -> getContentItemsTotal()
-                FAILED -> 1
-                EMPTY -> 1
+                State.LOADING -> 1
```
I'm wondering if grouping the same actions is more readable. Again nothing to be changed here, just a preference, but IMHO when reading a when statement is easy to understand the login by knowing the different paths we can take. Something like:
```
val contentTotal: Int = when (state) {
   State.LOADING, State.FAILED, State.EMPTY -> 1,
   State.LOADED -> getContentItemsTotal(),
   else -> throw IllegalStateException("Invalid mState")
}
```

ParaskP7 replied:
👋 @atorresveiga and thank you for the comment! 🙇

Same [response](https://github.com/woocommerce/woocommerce-android/pull/6865#discussion_r916323267)! 😅

Btw, the `else -> throw IllegalStateException("Invalid mState")` is redundant in the above example, I am not sure if you added that on purpose (see ccda8a62213ca2b9e604f4cd5851452dc60c7784 commit). 😅

atorresveiga commented on WooCommerce/src/main/kotlin/com/woocommerce/android/widgets/sectionedrecyclerview/Section.kt:
```diff
@@ -142,10 +139,10 @@ abstract class Section(sectionParameters: SectionParameters) {
-            LOADED -> onBindItemViewHolder(holder, position)
-            FAILED -> onBindFailedViewHolder(holder)
-            EMPTY -> onBindEmptyViewHolder(holder)
+            State.LOADING -> onBindLoadingViewHolder(holder)
```
Same as above.

ParaskP7 replied:
👋 @atorresveiga and thank you for the comment! 🙇

Same [response](https://github.com/woocommerce/woocommerce-android/pull/6865#discussion_r916658368)! 😅

atorresveiga commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/refunds/IssueRefundViewModel.kt:
```diff
@@ -959,7 +959,7 @@ class IssueRefundViewModel @Inject constructor(
             get() = grandTotalRefund > BigDecimal.ZERO
 
-        val isRefundNoticeVisible = !refundNotice.isNullOrEmpty()
+        @Suppress("PROPERTY_WONT_BE_SERIALIZED") val isRefundNoticeVisible = !refundNotice.isNullOrEmpty()
```
I wonder if we should add a TODO comment here or create an issue on Github just to ensure we don't forget to work on this in the future.

ParaskP7 replied:
👋 @atorresveiga !

Good call, I agree, this is now done here: 700b07635d86f6fa1efa17427f3c35b53a645b26

Related Issue: https://github.com/woocommerce/woocommerce-android/issues/6895

atorresveiga reviewed (approved):
Great work solving this technical debt @ParaskP7 🙌 . Changes looks good to me. I also smoke test the app an everything is working as expected. I left a few comments but nothing that requires a change. I'll approve this and you can merge regardless of your decision about the comments.

ParaskP7 committed:
TODO: Add todo comment with an assignee to resolve the suppress warning

Issue: Resolve suppressed 'PROPERTY_WONT_BE_SERIALIZED' warning for
'RefundByItemsViewState'
Link: https://github.com/woocommerce/woocommerce-android/issues/6895

ParaskP7 committed:
Merge branch 'trunk' of github.com:woocommerce/woocommerce-android into analysis/woo-commerce-type-cast-when-warnings

 Conflicts:
	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/cardreader/onboarding/CardReaderOnboardingChecker.kt

ParaskP7 commented:
👋 @atorresveiga !

> Changes looks good to me. I also smoke test the app an everything is working as expected.

Thank you so much for reviewing and smoke testing the app! 🙇 

> I left a few comments but nothing that requires a change.

Thank you so much for the comment, I tried to response in as much detail as possible to give you more context on why I chose one solution over the other, I hope it helps. 🙏 

>  I'll approve this and you can merge regardless of your decision about the comments.

Thank you, I applied one of your comment on create a `TODO`. About the `when` comments, I believe it will be good if we proceed as is right now, just so that to not block the progress on the overall `warnings-as-errors` project, we are so close!

But, if you feel strongly about that, please feel free to update those at any point to your preferred style, I am totally fine with that. 💯

atorresveiga commented:
Thanks, @ParaskP7 for the responses explaining in detail your mental mode when using the `else` branch. It makes perfect sense to me, and as I mentioned before, there was nothing that required a change. Great work here 🙌
